### PR TITLE
Delegate to :ssl for setopts/2 when transport is tls

### PR DESCRIPTION
**Pull request by starbelly**
Previously we always called `:inet.setopts/2` when a timeout was detected to set `{linger, {true, 0}}` to immediately tear down the socket before closing the connection. In doing so and in the case of tls the sslsocket structure gets passed to `:prim_inet.setopts/2` where the first argument is expected to be a port identifier.

### CODE REVIEW

File: test/client_and_receiver_integration_test.exs [hunk 1]
```diff
@@ -580,7 +580,7 @@ defmodule ClientAndReceiverIntegrationTest do
     @tag client_cert: "tls/server/server_certificate.pem"
     @tag keyfile: "tls/server/private_key.pem"
 
-    @tag reason: [:handshake_failure]
+    @tag reason: [:unsupported_certificate]
```
Comment (vikas15bhardwaj) [inline test/client_and_receiver_integration_test.exs:583]: this test is failing on my local. I also reverted this change. It does not return an error

**Comment by vikas15bhardwaj** [comment]
should we bump version?

File: test/client_and_receiver_integration_test.exs (hunk 1 above)
Comment (vikas15bhardwaj) [reply test/client_and_receiver_integration_test.exs:583]: I think erlang version should be `26.2.5.6`, test worked after I changed to this version

**Comment by vikas15bhardwaj** [comment]
In WP we are using elixir `1.15.7` version, won't we need to update WP elixir version before using this version of `mllp`?

File: test/client_and_receiver_integration_test.exs (hunk 1 above)
Comment (starbelly) [reply test/client_and_receiver_integration_test.exs:583]: it should be 😄

**Comment by starbelly** [comment]
> In WP we are using elixir `1.15.7` version, won't we need to update WP elixir version before using this version of `mllp`?

That won't be needed.

**Comment by starbelly** [comment]
> should we bump version?

We can do that.

**Comment by vikas15bhardwaj** [comment]
> > In WP we are using elixir `1.15.7` version, won't we need to update WP elixir version before using this version of `mllp`?
> 
> That won't be needed.

how does this work? will `mllp` be compiled to `1.15.7` version within WP?

**Pull request merged**